### PR TITLE
Allow arbitrary coordinate systems for RaycastVehicle.

### PR DESCRIPTION
**Pull request by nick70**
This is useful for vehicles that need to be reoriented in order to drive correctly.

### CODE REVIEW

File: Source/Urho3D/Physics/RaycastVehicle.h [hunk 1]
```diff
@@ -168,6 +168,8 @@ class URHO3D_API RaycastVehicle : public LogicComponent
     WeakPtr<RigidBody> hullBody_;
     /// Opaque Bullet data hidden from public
     RaycastVehicleData* vehicleData_;
+    /// Coordinate system
```
Comment (weitjong) [inline Source/Urho3D/Physics/RaycastVehicle.h]: Terminate the doxygen comment with a period. Also, may be declare an enumeration or constants for the coordinate system. They should be more readable than 0, 1, and 2.

**Comment by weitjong** [comment]
Thanks for your PR.

**Comment by nick70** [comment]
I think that should work. Changes in the attribute won't take effect until SetCoordinateSystem gets called again. It gets called eventually in ApplyAttributes.

File: Source/Urho3D/Physics/RaycastVehicle.cpp [hunk 2]
```diff
@@ -670,6 +677,17 @@ float RaycastVehicle::GetInAirRPM() const
     return inAirRPM_;
 }
 
+void RaycastVehicle::SetCoordinateSystem(const IntVector3& coordinateSystem)
+{
+    coordinateSystem_ = coordinateSystem;
+    vehicleData_->SetCoordinateSystem(coordinateSystem_);
+}
+
+IntVector3 RaycastVehicle::GetCoordinateSystem()
```
Comment (weitjong) [inline Source/Urho3D/Physics/RaycastVehicle.cpp]: The getter should be declared as `const`.

Comment (weitjong) [reply Source/Urho3D/Physics/RaycastVehicle.cpp]: And move to header.

File: Source/Urho3D/Physics/RaycastVehicle.cpp [hunk 3]
```diff
@@ -34,6 +34,8 @@
 namespace Urho3D
 {
 
+const IntVector3 RaycastVehicle::DEFAULT_COORDINATE_SYSTEM(0, 1, 2);
```
Comment (weitjong) [inline Source/Urho3D/Physics/RaycastVehicle.cpp]: Since 3P3 = 6, do you think it makes sense to declare all the possible permutations here?

**Comment by weitjong** [comment]
This is merged now. Thanks again.

**Pull request closed without merging**